yarn/history/src/main/scala/org/apache/spark/deploy/history/yarn/YarnHistoryService.scala

```diff
@@ -0,0 +1,981 @@
+        eventHandlingThread.foreach(_.interrupt())
+      }
+      stopTimelineClient
+      logInfo(s"Stopped: $this")
+    }
+  }
+
+  /**
+   * Get the current entity, creating one on demand if needed, setting up its entity Id
+   * from the current appID and attempt ID values, filters from the current name and user
+   * @return the current entity
+   */
+  private def getOrCreateCurrentEntity() = {
+    curEntity.getOrElse {
+      currentEventCount = 0
+      val entity = createTimelineEntity(
+        _applicationId,
+        _attemptId,
+        sparkApplicationId,
+        sparkApplicationAttemptId,
+        appStartEventProcessed.get(),
+        appName,
+        userName)
+      curEntity = Some(entity)
+      logDebug(s"Demand creation of new entity ${describeEntity(entity)}")
+      entity
+    }
+  }
+
+  /**
+   * Perform any preflight checks.
+   * @param entity timeline entity to review.
+   */
+  private[yarn] def preflightCheck(entity: TimelineEntity): Unit = {
+    require(entity.getStartTime != null,
+      s"No start time in ${describeEntity(entity)}")
+  }
+
+  /**
+   * If there is any available entity to be sent, push to timeline server
+   */
+  private def flushEntity(): Unit = {
+    if (entityList.nonEmpty) {
+      val count = flushCount.incrementAndGet()
+      logDebug(s"flushEntity #$count: list size ${entityList.size}")
+      var client = getTimelineClient
+      val lastUpdated = now()
+      // attempt to post each entity, removing from the list all of those
+      // which were successfully posted. The remainder are retained for
+      // followon attempts
+      entityList = entityList.filter { entity => {
+          if (entity == null) {
+            false
+          } else {
+            entity.addOtherInfo(FIELD_LAST_UPDATED, lastUpdated)
+            if (domainId != null) {
+              entity.setDomainId(domainId)
+            }
+            val entityDescription = describeEntity(entity)
+            logDebug(s"About to POST $entityDescription")
+            try { 
+              val response: TimelinePutResponse = client.putEntities(entity)
+              val errors = response.getErrors
+              if (!response.getErrors.isEmpty) {
+                val err = errors.get(0)
+                eventPostFailures.addAndGet(errors.size())
+                if (err.getErrorCode != 0) {
+                  logError(s"Failed to post ${entityDescription}\n:${describeError(err)}")
+                }
+              } else {
+                // successful submission
+                logDebug(s"entity successfully posted: $entityDescription")
+              }
+              // whatever the outcome, this request is not re-issued
+              false
+            } catch {
+              case e: ConnectException =>
+                eventPostFailures.incrementAndGet
+                logWarning(s"Connection exception submitting $entityDescription\n$e", e)
```

Comment:
OK. Historical habit from fielding support issues related to log traces of loggers set up to not log the stack trace.